addons/mail/static/src/models/composer/composer.js

```diff
@@ -118,6 +120,35 @@ registerModel({
                 textInputSelectionDirection: clear(),
             });
         },
+        async RPClinkPreview(url) {
+            const attachment = await this.env.services.rpc({
+                route: '/mail/link_preview',
+                params: { url, channel_id: this.activeThread.id },
+            }, { shadow: true });
+
+            if (!attachment) {
+                return;
+            }
+            this.update({ attachments: insert(attachment) });
+        },
```

Comment:
Isolate the RPC and return in a `modelMethod` named `performX` (e.g. `performLinkPreview`).
This is consistent with other uses of RPC calls.

Reply:
Other performX method do not isolate the RPC call. 
See https://github.com/odoo-dev/odoo/blob/master-mail-implement_opengraph_attachment-did/addons/mail/static/src/models/thread/thread.js#L489

Reply:
Making this a `modelMethods`is really annoying, since you can't access `this.activeThread` and `this.update`.
I don't understand this change.

Reply:
At leat the rename them and ok to keep it as a record method.